src/modules/Packages.rb

```diff
@@ -2420,6 +2355,42 @@ def remote_x11_packages
 
   private
 
+    def fetch_selected(category)
+      items = Pkg.ResolvableProperties("", category, "").select { |i| i["status"] == :selected }
+      items.each_with_object([]) { |i, a| a << i["name"] }.sort
```

Comment:
Ah, after changing `map` to `select` in the line above you do not need `each_with_object` here:

```suggestion
      items.map{ |i| i["name"] }.sort
```

Reply:
NP: @lslezak , yes, but using your `.each_with_object` suggestion the item names could be collected in a single iteration (without `nil`s) and then just sort them.

```ruby
result = []

Pkg.ResolvableProperties("", category, "").each_with_object(result) do |item|
  result << item["name"] if item["status"] == :selected
end

result.sort
```